libs/blocks/global-navigation/global-navigation.js

```diff
@@ -68,69 +70,137 @@ class Gnav {
             ${this.decorateMainNav()}
             ${this.decorateSearch()}
           </div>
-          ${this.decorateProfile()}
+          ${this.blocks.profile.blockEl && this.blocks.profile.decoratedEl}
           ${this.decorateLogo()}
         </nav>
         ${this.decorateBreadcrumbs()}
       </div>
     `;
     this.el.addEventListener('click', this.loadDelayed);
     setTimeout(() => this.loadDelayed(), 3000);
+    this.loadIMS();
     this.el.append(this.curtain, nav);
   };
 
   loadDelayed = async () => {
+    // eslint-disable-next-line no-async-promise-executor
     this.ready = this.ready || new Promise(async (resolve) => {
       this.el.removeEventListener('click', this.loadDelayed);
       const [
         { MenuControls },
         { decorateMenu, decorateLargeMenu },
         { appLauncher },
-        { profile },
+        { Profile },
         { Search },
       ] = await Promise.all([
         loadBlock('./delayed-utilities.js'),
         loadBlock('./blocks/navMenu/menu.js'),
         loadBlock('./blocks/appLauncher/appLauncher.js'),
         loadBlock('./blocks/profile/profile.js'),
         loadBlock('./blocks/search/gnav-search.js'),
+        loadStyles('profile/profile.css'),
         loadStyles('navMenu/menu.css'),
         loadStyles('search/gnav-search.css'),
       ]);
       this.menuControls = new MenuControls();
       this.decorateMenu = decorateMenu;
       this.decorateLargeMenu = decorateLargeMenu;
       this.appLauncher = appLauncher;
-      this.profile = profile;
+      this.Profile = Profile;
       this.search = Search;
-
-      this.imsReady
-        .then(({ blockEl, profileEl }) => {
-          this.decorateProfileMenu(blockEl, profileEl);
-        });
       resolve();
     });
     return this.ready;
   };
 
-  decorateProfileMenu = async (blockEl, profileEl) => {
+  loadIMS = () => {
+    const { locale, imsClientId, env } = getConfig();
+    if (!imsClientId) return null;
+    window.adobeid = {
+      client_id: imsClientId,
+      scope: 'AdobeID,openid,gnav',
+      locale: locale || 'en-US',
+      autoValidateToken: true,
+      environment: env.ims,
+      useLocalStorage: false,
+      onReady: () => this.decorateProfile(),
+    };
+    const imsScriptEl = document.querySelector('script[src$="/imslib.min.js"]') instanceof HTMLElement;
```

Comment:
Just a nitpick, the naming suggests this variable holds an element, when in fact it holds a boolean value of whether the element is defined.